clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -5760,6 +5776,10 @@ void CXXNameMangler::mangleValueInTemplateArg(QualType T, const APValue &V,
     mangleFloatLiteral(T, V.getFloat());
     break;
 
+  case APValue::DecimalFloat:
+    mangleType(T);
```

Comment:
Mangling of the value will be necessary here as well. See https://godbolt.org/z/3Edvh7xa4.